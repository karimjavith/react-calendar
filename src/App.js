import React, { Component } from "react";
import { Provider } from "react-redux";
import { HashRouter, Route, Redirect, Switch } from "react-router-dom";
import Media from "react-media";
import configureStore from "./redux/store";
import MonthCalendarContainer from "./components/Calendar/container/Month.Calendar.Container";
import DayCalendarContainer from "./components/Calendar/container/Day.Calendar.Container";
import WeekCalendarContainer from "./components/Calendar/container/Week.Calendar.Container";
import DayWeekHeaderContainer from "./components/Calendar/container/DayWeekHeader.Container";
import MyCalendarContainer from "./components/Calendar/container/MyCalendar.Container";
import ErrorBoundaryContainer from "./components/Shared/errorBoundary/ErrorBoundary.Container";
import ModalContainer from "./components/Shared/modal/modal.container";
import "./App.scss";

const store = configureStore();
class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <HashRouter>
          <div className="main-content container-fluid">
            <div className="main-content-header-container">
              <div className="header-descr">
                <p>Event calendar</p>
              </div>
            </div>
            <div className="main-content-container row">
              <div className="sub-calendar-container col-xs-12 col-sm-12 col-md-3 col-lg-3">
                <div className="sub-calendar-container__wrapper row">
                  <div className="sub-calendar-container__month hidden-xs hidden-sm col-md-12 col-lg-12">
                    <MonthCalendarContainer />
                  </div>
                  <div className="sub-calendar-container__mycalendar col-xs-12 col-sm-12 col-md-12 col-lg-12">
                    <ErrorBoundaryContainer>
                      <MyCalendarContainer />
                    </ErrorBoundaryContainer>
                  </div>
                </div>
              </div>
              <div className="sub-content-container col-xs-12 col-sm-12 col-md-9 col-lg-9">
                <ErrorBoundaryContainer>
                  <DayWeekHeaderContainer />
                </ErrorBoundaryContainer>
                <Media query="(max-width: 599px)">
                  {screenIsSmall =>
                    screenIsSmall ? (
                      // small screen has no redirect
                      <Switch>
                        <Route
                          path="/dayCalendar"
                          render={() => {
                            return (
                              <ErrorBoundaryContainer>
                                <DayCalendarContainer />
                              </ErrorBoundaryContainer>
                            );
                          }}
                        />
                        <Redirect from="/" to="/dayCalendar" />
                      </Switch>
                    ) : (
                        // large screen does!
                        <Switch>
                          <Route
                            path="/dayCalendar"
                            render={() => {
                              return (
                                <ErrorBoundaryContainer>
                                  <DayCalendarContainer />
                                </ErrorBoundaryContainer>
                              );
                            }}
                          />
                          <Route
                            path="/weekCalendar"
                            render={() => {
                              return (
                                <ErrorBoundaryContainer>
                                  <WeekCalendarContainer />
                                </ErrorBoundaryContainer>
                              );
                            }}
                          />
                          <Redirect from="/" to="/dayCalendar" />
                        </Switch>
                      )
                  }
                </Media>
              </div>
            </div>
            <ModalContainer />
          </div>
        </HashRouter>
      </Provider>
    );
  }
}

export default App;
