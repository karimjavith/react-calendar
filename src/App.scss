.rdc-container {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto',
    'Helvetica Neue', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  line-height: 1rem;
  letter-spacing: 1px;
  font-weight: 400;
  display: flex;
  flex-direction: column;
  width: 100%;
  color: #1c1c1e;
  > *,
  ::after,
  ::before {
    box-sizing: border-box;
  }
  &__btn {
    cursor: pointer;
    padding: 1rem;
    background: #333335;
    text-decoration: none;
    user-select: none;
    font-size: 1rem;
    &:hover {
      background: #fff;
      color: #333335;
    }
    &--today {
      border: 1px solid #333335;
      background: transparent;
      width: 100px;
      text-transform: uppercase;
      &:hover {
        background: #333335;
        color: #fff;
      }
    }
  }
  &__header {
    display: inline-block;
    width: 100%;
    padding: 1rem;
    @media screen and (max-device-width: 767px) {
      display: flex;
      flex-direction: column;
      &__group {
        .rdc-container__btn--today {
          width: 100%;
          margin-bottom: 1rem;
        }
        &:not(:first-child) {
          padding: 1rem 0;
        }
        justify-content: center;
      }
    }
    &__group {
      display: inline-flex;
      &:not(:first-child) {
        margin: 0 1rem;
      }
      &__item {
        padding: 0 1rem;
      }
    }
  }
  &__row {
    display: flex;
    padding: 1rem;
    width: 100%;
    @media screen and (max-device-width: 767px) {
    }
    &__column {
      @media screen and (max-device-width: 767px) {
        margin-right: 0.5rem;
      }
      &--left {
        margin-right: 1rem;
      }
      &--right {
        width: 100%;
        &::after {
          content: '';
          display: block;
          border-bottom: 1px solid #ddd;
          width: 100%;
          height: 1px;
          margin-top: 8px;
        }
      }
      &__event {
        &__content {
          padding: 0.5rem;
        }
        position: absolute;
        border-radius: 6px;
      }
    }
  }
}
