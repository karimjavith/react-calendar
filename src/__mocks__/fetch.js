window.fetch = jest.fn().mockImplementationOnce(() => {
  return new Promise((res, rej) => res({}));
});
