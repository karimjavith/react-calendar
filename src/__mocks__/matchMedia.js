export function mockMatchMedia(shouldMatch, mediaQuery) {
  return (window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: shouldMatch,
      media: mediaQuery,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn()
    };
  }));
}
