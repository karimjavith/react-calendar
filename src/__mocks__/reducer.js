export function reducer(state, action) {
  return {
    ...state,
    calendar: {
      isFetching: false
    }
  };
}
