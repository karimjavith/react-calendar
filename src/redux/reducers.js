import { combineReducers } from "redux";
import calendar from "../components/Calendar/store";
import modal from '../components/Shared/modal/store/modal.store';

function getReducer(initialState, handlers) {
  return function reducer(state = initialState, action) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action)
    }
    return state
  }
}
export default function createReducer() {
  return combineReducers({
    calendar: getReducer(calendar.initialState, calendar.handlers),
    modal: getReducer(modal.initialState, modal.handlers)
  });
}
