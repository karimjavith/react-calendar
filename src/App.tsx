/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import dateFns from 'date-fns';
import DayWeekHeader from './components/calendar/DayWeekHeader';
import Day from './components/calendar/day/index';
import './App.scss';
import { Events } from './type';
type AppProps = {
  selectedDate: string;
  events: Events[];
  isLoading: boolean;
};
const App = (props: AppProps) => {
  const initialState = {
    selectedDate: props.selectedDate
  };
  const [state, setstate] = useState(initialState);
  const changeDate = (by: number) => {
    setstate({
      ...state,
      selectedDate:
        by !== 0
          ? dateFns.addDays(state.selectedDate, by).toLocaleDateString()
          : props.selectedDate
    });
  };
  return (
    <div className="rdc-container">
      <DayWeekHeader
        selectedDate={state.selectedDate}
        changeDate={changeDate}
      />
      <Day
        events={props.events}
        isLoading={props.isLoading}
        selectedDate={state.selectedDate}
      />
    </div>
  );
};
App.displayName = 'Day Calendar';
export default App;
