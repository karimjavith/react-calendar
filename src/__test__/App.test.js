//__test__/App.test.js
import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import {
  render,
  cleanup,
  wait,
} from "react-testing-library";
import {
  createHistory,
  createMemorySource,
  LocationProvider
} from "@reach/router";
import "jest-dom/extend-expect";
import { reducer } from "../__mocks__/reducer";
import { mockMatchMedia } from "../__mocks__/matchMedia";
import App from "../App";

function renderWithReduxAndRouter(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
  { route = "/", history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <LocationProvider history={history}>{ui}</LocationProvider>
      </Provider>
    ),
    store,
    history
  };
}

test("should render <App /> component with day view", async () => {
  mockMatchMedia(false, "max-width: 1200px");
  const {
    container,
    getByText,
    history: { navigate }
  } = renderWithReduxAndRouter(<App />, {
    initialState: {
      calendar: { isFetching: false }
    }
  });
  await wait().then(jest.runOnlyPendingTimers());
  expect(getByText("Event calendar")).toBeInTheDocument();
  expect(getByText("My Calendars")).toBeInTheDocument();

  expect(container.firstChild).toMatchSnapshot();
});

test("should render <App /> component with week view", async () => {
  mockMatchMedia(true, "max-width: 1200px");
  const {
    container,
    getByText,
    history: { navigate }
  } = renderWithReduxAndRouter(<App />, {
    route: "/weekCalendar",
    initialState: {
      calendar: { isFetching: false }
    }
  });
  await wait().then(jest.runOnlyPendingTimers());
  expect(
    container.querySelectorAll(".week-calendar-container")
  ).toMatchSnapshot();
});
test("should render <App /> component with day view for smaller devices", async () => {
  mockMatchMedia(true, "max-width: 400px");
  const {
    container,
    history: { navigate }
  } = renderWithReduxAndRouter(<App />, {
    initialState: {
      calendar: { isFetching: false }
    }
  });
  await wait().then(jest.runOnlyPendingTimers());
  expect(
    container.querySelector(".day-calendar-container")
  ).toMatchSnapshot();
});
