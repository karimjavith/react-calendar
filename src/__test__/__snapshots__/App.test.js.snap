// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <App /> component with day view 1`] = `
<div
  class="main-content container-fluid"
>
  <div
    class="main-content-header-container"
  >
    <div
      class="header-descr"
    >
      <p>
        Event calendar
      </p>
    </div>
  </div>
  <div
    class="main-content-container row"
  >
    <div
      class="sub-calendar-container col-xs-12 col-sm-12 col-md-3 col-lg-3"
    >
      <div
        class="sub-calendar-container__wrapper row"
      >
        <div
          class="sub-calendar-container__month hidden-xs hidden-sm col-md-12 col-lg-12"
        >
          <div
            class="calendar"
          >
            <div
              class="header row flex-middle"
            >
              <div
                class="col col-start"
              >
                <div
                  class="icon"
                >
                  <span>
                    <i
                      class="glyphicon glyphicon-chevron-left"
                    />
                  </span>
                </div>
              </div>
              <div
                class="col col-center"
              >
                <span>
                  February 2019
                </span>
              </div>
              <div
                class="col col-end"
              >
                <div
                  class="icon"
                >
                  <span>
                    <i
                      class="glyphicon glyphicon-chevron-right"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="days row"
            >
              <div
                class="col col-center"
              >
                S
              </div>
              <div
                class="col col-center"
              >
                M
              </div>
              <div
                class="col col-center"
              >
                T
              </div>
              <div
                class="col col-center"
              >
                W
              </div>
              <div
                class="col col-center"
              >
                T
              </div>
              <div
                class="col col-center"
              >
                F
              </div>
              <div
                class="col col-center"
              >
                S
              </div>
            </div>
            <div
              class="body"
            >
              <div
                class="row"
              >
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="27"
                  >
                    27
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="28"
                  >
                    28
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="29"
                  >
                    29
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="30"
                  >
                    30
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="31"
                  >
                    31
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="1"
                  >
                    1
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="2"
                  >
                    2
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="3"
                  >
                    3
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="4"
                  >
                    4
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="5"
                  >
                    5
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="6"
                  >
                    6
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="7"
                  >
                    7
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="8"
                  >
                    8
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="9"
                  >
                    9
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="10"
                  >
                    10
                  </span>
                </div>
                <div
                  class="col cell selected"
                >
                  <span
                    class="number"
                    data-formatted="11"
                  >
                    11
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="12"
                  >
                    12
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="13"
                  >
                    13
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="14"
                  >
                    14
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="15"
                  >
                    15
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="16"
                  >
                    16
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="17"
                  >
                    17
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="18"
                  >
                    18
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="19"
                  >
                    19
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="20"
                  >
                    20
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="21"
                  >
                    21
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="22"
                  >
                    22
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="23"
                  >
                    23
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="24"
                  >
                    24
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="25"
                  >
                    25
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="26"
                  >
                    26
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="27"
                  >
                    27
                  </span>
                </div>
                <div
                  class="col cell "
                >
                  <span
                    class="number"
                    data-formatted="28"
                  >
                    28
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="1"
                  >
                    1
                  </span>
                </div>
                <div
                  class="col cell disabled"
                >
                  <span
                    class="number"
                    data-formatted="2"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sub-calendar-container__mycalendar col-xs-12 col-sm-12 col-md-12 col-lg-12"
        >
          <div
            class="my-calendar-container"
          >
            <div
              class="my-calendar-container__header"
            >
              <span>
                <h4>
                  My Calendars
                </h4>
              </span>
            </div>
            <div
              class="my-calendar-container__list-container"
            >
              <ul
                class="my-calendar-container__list row"
              >
                <li>
                  <div
                    class="checkbox-containercol-sm-12 col-md-12"
                  >
                    <label
                      class="custom-control-checkbox-container"
                    >
                      <input
                        aria-label="checkbox for calendar types"
                        class="mycalendar-checkbox-input custom-control-input"
                        id="checkbox-calendar-type1"
                        type="checkbox"
                      />
                      <span
                        class="custom-control-indicator"
                      />
                      <span
                        class="mycalendar-checkbox-label custom-control-description"
                        for="checkbox-calendar-type1"
                      >
                        Calendar 1
                      </span>
                    </label>
                  </div>
                </li>
                <li>
                  <div
                    class="checkbox-containercol-sm-12 col-md-12"
                  >
                    <label
                      class="custom-control-checkbox-container"
                    >
                      <input
                        aria-label="checkbox for calendar types"
                        class="mycalendar-checkbox-input custom-control-input"
                        id="checkbox-calendar-type1"
                        type="checkbox"
                      />
                      <span
                        class="custom-control-indicator"
                      />
                      <span
                        class="mycalendar-checkbox-label custom-control-description"
                        for="checkbox-calendar-type1"
                      >
                        Calendar 2
                      </span>
                    </label>
                  </div>
                </li>
                <li>
                  <div
                    class="checkbox-containercol-sm-12 col-md-12"
                  >
                    <label
                      class="custom-control-checkbox-container"
                    >
                      <input
                        aria-label="checkbox for calendar types"
                        class="mycalendar-checkbox-input custom-control-input"
                        id="checkbox-calendar-type1"
                        type="checkbox"
                      />
                      <span
                        class="custom-control-indicator"
                      />
                      <span
                        class="mycalendar-checkbox-label custom-control-description"
                        for="checkbox-calendar-type1"
                      >
                        Calendar 3
                      </span>
                    </label>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sub-content-container col-xs-12 col-sm-12 col-md-9 col-lg-9"
    >
      <div
        class="sub-content-container__header row"
      >
        <div
          class="sub-content-container__header-dateTimeDisplay  col-xs-12 col-sm-12 col-md-8 "
        >
          <div
            class="row"
          >
            <div
              class="col-md-2"
            >
              <span>
                <i
                  class="glyphicon glyphicon-menu-left"
                />
              </span>
              <span>
                <i
                  class="glyphicon glyphicon-menu-right"
                />
              </span>
            </div>
            <div
              class="col-md-10"
            >
              11 February 2019
            </div>
          </div>
        </div>
        <div
          class="sub-content-container__header-navMenu hidden-xs hidden-sm col-md-4 col-lg-4"
        >
          <ul
            class="day-week-calendar-menu-container"
          >
            <div
              class="day-calendar-menu-container"
            >
              <ul
                class="nav nav-pills"
              >
                <li
                  class="active"
                >
                  <a
                    aria-current="page"
                    class="active"
                    href="#/dayCalendar"
                  >
                    Day
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="week-calendar-menu-container"
            >
              <ul
                class="nav nav-pills"
              >
                <li
                  class=""
                >
                  <a
                    class=""
                    href="#/weekCalendar"
                  >
                    Week
                  </a>
                </li>
              </ul>
            </div>
          </ul>
        </div>
         
      </div>
      <div
        class="day-calendar-container"
      >
        <div
          class="one"
        >
          <div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                00
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                01
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                02
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
                <div
                  class="event-container-wrapper"
                >
                  <div>
                    <div
                      class="event-container"
                    >
                      <div
                        class="event-card"
                        role="button"
                        style="top: 120px; height: 80px; width: 100%;"
                      >
                        <div
                          class="event-name"
                        >
                          <span>
                            Event Name
                          </span>
                        </div>
                        <div
                          class="event-time"
                        >
                          <span>
                            12:30 - 14:30
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                03
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                04
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                05
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                06
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                07
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                08
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                09
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
                <div
                  class="event-container-wrapper"
                >
                  <div>
                    <div
                      class="event-container"
                    >
                      <div
                        class="event-card"
                        role="button"
                        style="top: 400px; height: 40px; width: 100%;"
                      >
                        <div
                          class="event-name"
                        >
                          <span>
                            Event Name
                          </span>
                        </div>
                        <div
                          class="event-time"
                        >
                          <span>
                            12:30 - 14:30
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                10
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                11
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                12
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                13
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                14
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                15
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                16
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                17
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                18
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                19
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                20
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                21
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                22
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
                <div
                  class="same-hour-container"
                  style="top: 20px;"
                />
              </div>
            </div>
            <div
              class="hour-container-wrapper"
            >
              <div
                class="hour"
              >
                23
                :00
              </div>
              <div
                class="hour-inner"
              >
                <div
                  class="border-bottom"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  
</div>
`;

exports[`should render <App /> component with day view for smaller devices 1`] = `
<div
  class="day-calendar-container"
>
  <div
    class="one"
  >
    <div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          00
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          01
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          02
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
          <div
            class="event-container-wrapper"
          >
            <div>
              <div
                class="event-container"
              >
                <div
                  class="event-card"
                  role="button"
                  style="top: 120px; height: 80px; width: 100%;"
                >
                  <div
                    class="event-name"
                  >
                    <span>
                      Event Name
                    </span>
                  </div>
                  <div
                    class="event-time"
                  >
                    <span>
                      12:30 - 14:30
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          03
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          04
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          05
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          06
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          07
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          08
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          09
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
          <div
            class="event-container-wrapper"
          >
            <div>
              <div
                class="event-container"
              >
                <div
                  class="event-card"
                  role="button"
                  style="top: 400px; height: 40px; width: 100%;"
                >
                  <div
                    class="event-name"
                  >
                    <span>
                      Event Name
                    </span>
                  </div>
                  <div
                    class="event-time"
                  >
                    <span>
                      12:30 - 14:30
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          10
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          11
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          12
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          13
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          14
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          15
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          16
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          17
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          18
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          19
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          20
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          21
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          22
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
          <div
            class="same-hour-container"
            style="top: 20px;"
          />
        </div>
      </div>
      <div
        class="hour-container-wrapper"
      >
        <div
          class="hour"
        >
          23
          :00
        </div>
        <div
          class="hour-inner"
        >
          <div
            class="border-bottom"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render <App /> component with week view 1`] = `
NodeList {
  Symbol(impl): NodeListImpl {
    "_isLive": false,
    "_list": Array [],
    Symbol(wrapper): [Circular],
  },
}
`;
